test/unix/test_lwt_bytes.ml

```diff
@@ -0,0 +1,497 @@
+        Lwt.return true
+      with
+      | Invalid_argument _ -> Lwt.return true
+      | _ -> Lwt.return false
+    end;
+
+    test "set out of bounds : upper limit" begin fun () ->
+      let buff = Lwt_bytes.create 3 in
+      try
+        let () = Lwt_bytes.set buff 150 'a' in
+        Lwt.return true
+      with
+      | Invalid_argument _ -> Lwt.return true
+      | _ -> Lwt.return false
+    end;
+
+    test "unsafe_get/unsafe_set" begin fun () ->
+      let buff = Lwt_bytes.create 4 in
+      let () = Lwt_bytes.unsafe_set buff 0 'a' in
+      let () = Lwt_bytes.unsafe_set buff 1 'b' in
+      let () = Lwt_bytes.unsafe_set buff 2 'c' in
+      let check = Lwt_bytes.unsafe_get buff 0 = 'a' &&
+                  Lwt_bytes.unsafe_get buff 1 = 'b' &&
+                  Lwt_bytes.unsafe_get buff 2 = 'c'
+      in Lwt.return check
+    end;
+
+    test "of bytes" begin fun () ->
+      let bytes = Bytes.of_string "abc" in
+      let buff = Lwt_bytes.of_bytes bytes in
+      let check = Lwt_bytes.get buff 0 = Bytes.get bytes 0 &&
+                  Lwt_bytes.get buff 1 = Bytes.get bytes 1 &&
+                  Lwt_bytes.get buff 2 = Bytes.get bytes 2
+      in Lwt.return check
+    end;
+
+    test "of string" begin fun () ->
+      let buff = Lwt_bytes.of_string "abc" in
+      let check = Lwt_bytes.get buff 0 = 'a' &&
+                  Lwt_bytes.get buff 1 = 'b' &&
+                  Lwt_bytes.get buff 2 = 'c'
+      in Lwt.return check
+    end;
+
+    test "to bytes" begin fun () ->
+      let bytes = Bytes.of_string "abc" in
+      let buff = Lwt_bytes.of_bytes bytes in
+      let bytes' = Lwt_bytes.to_bytes buff in
+      let check = bytes = bytes' in
+      Lwt.return check
+    end;
+
+    test "to string" begin fun () ->
+      let str = "abc" in
+      let buff = Lwt_bytes.of_string str in
+      let str' = Lwt_bytes.to_string buff in
+      let check = str = str' in
+      Lwt.return check
+    end;
+
+    test "blit" begin fun () ->
+      let str1 = "abc" in
+      let buf1 = Lwt_bytes.of_string str1 in
+      let str2 = "abcdef" in
+      let buf2 = Lwt_bytes.of_string str2 in
+      let () = Lwt_bytes.blit buf1 0 buf2 3 3 in
+      let check = "abcabc" = Lwt_bytes.to_string buf2 in
+      Lwt.return check
+    end;
+
+    test "blit source out of bounds: lower limit" begin fun () ->
+      let str1 = "abc" in
+      let buf1 = Lwt_bytes.of_string str1 in
+      let str2 = "abcdef" in
+      let buf2 = Lwt_bytes.of_string str2 in
+      try
+        let() = Lwt_bytes.blit buf1 (-1) buf2 3 3 in
+        Lwt.return_false
+      with
+      | Invalid_argument _ -> Lwt.return true
+      | _ -> Lwt.return false
+    end;
+
+    test "blit source out of bounds: upper limit" begin fun () ->
+      let str1 = "abc" in
+      let buf1 = Lwt_bytes.of_string str1 in
+      let str2 = "abcdef" in
+      let buf2 = Lwt_bytes.of_string str2 in
+      try
+        let() = Lwt_bytes.blit buf1 4 buf2 3 3 in
```

Comment:
Sorry, I made a mistake and didn't see the source is only 3 bytes (not 6), so I'd pick 1 or 2 here as the starting offset: the offset of a byte that is actually within the source buffer, so that it is adding the length which triggers the upper limit check failure.